pkg/sql/schemachanger/scop/ops.go

```diff
@@ -28,6 +28,10 @@ type Ops interface {
 // returns an implementation of Ops corresponding to that type. The set of ops
 // must all be the same type, otherwise MakeOps will panic.
 func MakeOps(ops ...Op) Ops {
+	// The type of stage doesn't matter for nil ops.
+	if len(ops) == 0 {
+		return nil
+	}
```

Comment:
I really wish I understood why this happens better. Can we eliminate this in the planning layer and avoid making this call?

Reply:
I'd be inclined to leave this be and revisit empty stages and other optimization questions at a later point in time, if that's OK with you.